src/shims/fs.rs

```diff
@@ -1103,6 +1113,81 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             this.handle_not_found()
         }
     }
+
+    fn fsync(&mut self, fd_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        // On macOS, fsync does not wait for the underlying disk to finish writing, while
+        // fcntl with the F_FULLFSYNC operation does. If fsync is somehow called by a macOS
+        // target, we will call File::sync_all(), which is a stronger operation that does
+        // wait for the disk to flush its buffers. (If the host is on Linux, the standard
+        // library will call fsync, as it does flush disk buffers, and if the host is
+        // is on macOS, it will call fcntl with F_FULLFSYNC.)
```

Comment:
```suggestion
        // on macOS, it will call fcntl with F_FULLFSYNC.)
```

Reply:
> I had meant that in the sense that both the target and the host were macOS, I'll clarify it.

Oh now I see what you mean, you are explaining what `sync_all` here does.
I don't think that's necessary, what matters is that `sync_all` is stronger than what the target requires. I'd just remove the parenthetical.

Also, "somehow called" is a bit odd, why not just "called"?